src/ApiResponse/AbstractApiJsonResponse.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Silktide\ProspectClient\ApiResponse;
+
+use Psr\Http\Message\ResponseInterface;
+use function GuzzleHttp\json_decode;
+
+class AbstractApiJsonResponse extends AbstractApiResponse
+{
+    /** @var object|array|null */
+    protected $jsonResponse;
+
+    public function __construct(ResponseInterface $httpResponse)
+    {
+        parent::__construct($httpResponse);
+        $body = $httpResponse->getBody();
+
+        if(is_null($body)) {
+            $this->jsonResponse = null;
+        }
+        else {
+            $this->jsonResponse = json_decode($body);
```

Comment:
Can you make this a `json_decode($body, true)` and use it as an array?

`stdClass` is a curse that should never have existed and doesn't deserve a place in the modern world :P